modules/gamAdServerSubmodule.js

```diff
@@ -0,0 +1,27 @@
+import { GAM_VENDOR } from './videoModule/constants/vendorCodes.js';
+import { adServerDirectory } from './videoModule/vendorDirectory.js';
+
+/**
+ * @constructor
+ * @param {Object} dfpModule_ - the DFP ad server module
+ * @returns {AdServerProvider}
+ */
+function GamAdServerProvider(dfpModule_) {
+  const dfp = dfpModule_;
+
+  function getAdTagUrl(adUnit, baseAdTag) {
+    return dfp.buildVideoUrl({ adUnit: adUnit, url: baseAdTag });
+  }
+
+  return {
+    getAdTagUrl
+  }
+}
+function gamSubmoduleFactory() {
+  const dfp = $$PREBID_GLOBAL$$.adServers.dfp;
```

Comment:
out of curiosity, why so ? I've noticed that the macro is used in a lot of modules; what are the differences between the two?

Reply:
The difference is that `getGlobal` is normal javascript, while `$$PREBID_GLOBAL$$` is substituted by a custom babel plugin. Ultimately the first is just a wrapper around the second so it should not matter, but since our custom logic is probably more likely to break than the javascript VM, I think it's good practice to keep it constrained to the minimum possible.